src/Message/CompleteRedirectPurchaseRequest.php

```diff
@@ -37,11 +39,29 @@ public function sendData($data): CompleteRedirectPurchaseResponse
         );
     }
 
+    /**
+     * @return string
+     * @throws InvalidRequestException
+     */
     public function getEndpoint(): string
     {
-        return sprintf($this->apiResourceString, parent::getEndpoint(),
+        $sessionID = $this->httpRequest->query->get('sessionId');
+        if (!$sessionID) {
+            // If this is from a webhook, we get the full transaction details in the post body
+            $json = json_decode($this->httpRequest->getContent(), true);
+            if ($json && !empty($json["sessionId"])) {
```

Comment:
@jamesgarton There is not need to test $json when you use empty function to test $json["sessionId"]